dotCMS/src/main/java/com/liferay/portal/ejb/PortletManagerUtil.java

```diff
@@ -1,186 +1,80 @@
-		}
-	}
-
-	public static void initEAR(java.lang.String[] xmls)
-		throws com.liferay.portal.SystemException {
-		try {
-			PortletManager portletManager = PortletManagerFactory.getManager();
-			portletManager.initEAR(xmls);
-		}
-		catch (Exception e) {
-			throw new com.liferay.portal.SystemException(e);
-		}
-	}
-
-	public static java.util.List initWAR(java.lang.String servletContextName,
-		java.lang.String[] xmls) throws com.liferay.portal.SystemException {
-		try {
-			PortletManager portletManager = PortletManagerFactory.getManager();
-
-			return portletManager.initWAR(servletContextName, xmls);
-		}
-		catch (Exception e) {
-			throw new com.liferay.portal.SystemException(e);
-		}
-	}
-
-	public static com.liferay.portal.model.Portlet updatePortlet(
-		java.lang.String portletId, java.lang.String groupId,
-		java.lang.String defaultPreferences, boolean narrow,
-		java.lang.String roles, boolean active)
-		throws com.liferay.portal.PortalException, 
-			com.liferay.portal.SystemException {
-		try {
-			PortletManager portletManager = PortletManagerFactory.getManager();
 
-			return portletManager.updatePortlet(portletId, groupId,
-				defaultPreferences, narrow, roles, active);
-		}
-		catch (com.liferay.portal.PortalException pe) {
-			throw pe;
-		}
-		catch (com.liferay.portal.SystemException se) {
-			throw se;
-		}
-		catch (Exception e) {
-			throw new com.liferay.portal.SystemException(e);
-		}
-	}
-}
\ No newline at end of file
+    public static Collection addPortlets(final java.lang.String[] xmls) throws com.liferay.portal.SystemException {
+        try {
+            final PortletFactory portletFactory = PortletManagerFactory.getManager();
+            
+            Map<String,Portlet> portlets = portletFactory.xmlToPortlets(xmls);
+            for(Portlet portlet : portlets.values()) {
+              portletFactory.insertPortlet(portlet);
+            }
+            
+            return portletFactory.getPortlets();
+        } catch (final Exception e) {
+            throw new com.liferay.portal.SystemException(e);
+        }
+    }
+
+    public static com.liferay.portal.model.Portlet getPortletById(final java.lang.String companyId, final java.lang.String portletId)
+            throws com.liferay.portal.SystemException {
+        try {
+            final PortletFactory portletFactory = PortletManagerFactory.getManager();
+
+            return portletFactory.findById(portletId);
+        } catch (final Exception e) {
+            throw new com.liferay.portal.SystemException(e);
+        }
+    }
+
+
+
+    @CloseDBIfOpened
+    public static java.util.Collection<Portlet> getPortlets(final java.lang.String companyId) throws com.liferay.portal.SystemException {
+        try {
+            final PortletFactory portletFactory = PortletManagerFactory.getManager();
+
+            return portletFactory.getPortlets();
+        } catch (final com.liferay.portal.SystemException se) {
+            throw se;

```

Comment:
I think this catch is not necessary because on the next lines we have
```
        } catch (final Exception e) {
            throw new com.liferay.portal.SystemException(e);
        }
```

Reply:
done